pkg/sql/distsqlrun/mergejoiner_test.go

```diff
@@ -553,3 +554,47 @@ func TestConsumerClosed(t *testing.T) {
 		})
 	}
 }
+
+func BenchmarkMergeJoiner(b *testing.B) {
+	ctx := context.Background()
+	evalCtx := tree.MakeTestingEvalContext()
+	defer evalCtx.Stop(ctx)
+	flowCtx := &FlowCtx{
+		Settings: cluster.MakeTestingClusterSettings(),
+		EvalCtx:  evalCtx,
+	}
+
+	spec := &MergeJoinerSpec{
+		LeftOrdering: convertToSpecOrdering(
+			sqlbase.ColumnOrdering{
+				{ColIdx: 0, Direction: encoding.Ascending},
+			}),
+		RightOrdering: convertToSpecOrdering(
+			sqlbase.ColumnOrdering{
+				{ColIdx: 0, Direction: encoding.Ascending},
+			}),
+		Type: JoinType_INNER,
+		// Implicit @1 = @2 constraint.
+	}
+	post := &PostProcessSpec{}
+
+	const numCols = 1
+	for _, numRows := range []int{0, 1 << 2, 1 << 4, 1 << 8, 1 << 12, 1 << 16} {
+		b.Run(fmt.Sprintf("rows=%d", numRows), func(b *testing.B) {
+			rows := makeIntRows(numRows, numCols)
+			leftInput := NewRepeatableRowSource(oneIntCol, rows)
+			rightInput := NewRepeatableRowSource(oneIntCol, rows)
+			b.SetBytes(int64(8 * numRows * numCols))
+			b.ResetTimer()
+			for i := 0; i < b.N; i++ {
+				h, err := newMergeJoiner(flowCtx, spec, leftInput, rightInput, post, &RowDisposer{})
```

Comment:
So I think the initialization of the processor here is pessimistically skewing the benchmark numbers.

Perhaps try pulling this out of the loop? I don't think any other state is mutated horribly in `mergeJoiner` after each successive `Run`.

Reply:
The initialization of the processor is only a factor at small `numRows`. And it is somewhat interesting to capture that initialization overhead. The hash joiner benchmark has a TODO about addressing this overhead, but I'm not convinced we want to change the benchmark.

Reply:
Fair enough 👍  the hash joiner's state with the memory containers also look non-trivial to address for the benchmark so it might be best to keep this uniform across the benchmarks.